transport_analysis/analysis/velocityautocorr.py

```diff
@@ -75,19 +76,29 @@ def __init__(
         # See more at the MDAnalysis documentation:
         # https://docs.mdanalysis.org/stable/documentation_pages/analysis/base.html?highlight=results#MDAnalysis.analysis.base.Results
 
-        # args
-        self.dim_type = dim_type
-        self._parse_dim_type()
-        # self.fft = fft
-
-        # local
-        self.atomgroup = atomgroup
-        self.n_particles = len(self.atomgroup)
-        self._velocity_array = None
-
-        # result
-        self.results.vacf_by_particle = None
-        self.results.timeseries = None
+        try:
+            if not atomgroup.universe.trajectory.has_velocities:
+                raise AttributeError
+        except AttributeError:
+            raise AttributeError("atomgroup must be from a trajectory with velocities")
+        else:
+            if isinstance(atomgroup, UpdatingAtomGroup):
+                raise TypeError("UpdatingAtomGroups are not valid for VACF "
+                                "computation")
+
+            # args
+            self.dim_type = dim_type
+            self._parse_dim_type()
+            self.fft = fft
+
+            # local
+            self.atomgroup = atomgroup
+            self.n_particles = len(self.atomgroup)
+            self._velocity_array = None
+
+            # result
+            self.results.vacf_by_particle = None
+            self.results.timeseries = None
```

Comment:
Maybe instantiate these as empty version of the data type you intend them to be?

Reply:
I do that in `_prepare()`, just like in the MSD module. Would it be better to do it here? Have not tried it myself

Reply:
Hmm, IMO it's nicer in terms of readability to have it only in one place, your call on where it goes. Defining empty versions is also nice because the types are clear.